update docs for for loop with conditional to reference filter filters

shieldo opened the pull request:
I don't know if there needs to be the same emphasis for this case as there's no special syntax for this any more? This PR assumes the same emphasis.

Commit:
update docs for for loop with conditional to reference filter filters

fabpot commented on doc/tags/for.rst:
```diff
@@ -85,13 +85,14 @@ Adding a condition
 
     <ul>
-        {% for user in users if user.active %}
+        {% for user in users|filter(v => v.active) %}
```
I would use `user` or `u` for the variable.

fabpot replied:
You should also add a note to say that `filter` is available as of Twig 1.41/2.10 (and mention the old syntax with the `if` condition).

fabpot commented:
Can you rebase on 1.x?

shieldo commented:
@fabpot Might it be better for me to submit another PR for the 1.x branch so I can reference 1.x releases there and 2.x releases here? I'm just looking at e.g. https://twig.symfony.com/doc/2.x/filters/filter.html#filter and it seems confusing/ possibly misleading that only a 1.x release is mentioned there in the 2.x docs (as the `filter` filter wasn't available in older 2.x releases).

fabpot commented:
You're right

shieldo commented:
@fabpot OK I changed my approach here and just added a tip for the deprecation/ replacement syntax, given it maybe doesn't strictly need a syntax example, and seemed more consistent with docs elsewhere (such as https://twig.symfony.com/doc/2.x/tags/spaceless.html). If that makes sense then I can submit a separate  PR for 1.x, and add a 3.x one that uses `filter` only.

fabpot commented on doc/tags/for.rst:
```diff
@@ -84,6 +84,13 @@ Variable              Description
+    As of Twig 2.10, use the :doc:`filter <../filters/filter>` filter instead,
+    or an ``if`` condition inside the ``for`` body (if your condition
+    depends on a variable updated inside the loop and you are not using the
+    `loop` variable).
```
double tick

shieldo replied:
Updated

fabpot commented:
@shieldo Works for me.

fabpot commented:
Thank you @shieldo.
